src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.props

```
@@ -72,6 +72,15 @@ Copyright (c) .NET Foundation. All rights reserved.
   </PropertyGroup>
 
   <!-- Default settings for all projects built with this Sdk package -->
+
+  <PropertyGroup Condition=" '$(AssemblySearchPaths)' == '' ">
+    <!-- By default exclude GAC, registry, output directory from search paths. -->
+    <AssemblySearchPaths Condition="$(AssemblySearchPathUseCandidateAssemblyFiles) != 'false'">{CandidateAssemblyFiles}</AssemblySearchPaths>
+    <AssemblySearchPaths Condition="$(AssemblySearchPathUseHintPathFromItem) != 'false'">$(AssemblySearchPaths);{HintPathFromItem}</AssemblySearchPaths>
+    <AssemblySearchPaths Condition="$(AssemblySearchPathUseTargetFrameworkDirectory) != 'false'">$(AssemblySearchPaths);{TargetFrameworkDirectory}</AssemblySearchPaths>
+    <AssemblySearchPaths Condition="$(AssemblySearchPathUseRawFileName) != 'false'">$(AssemblySearchPaths);{RawFileName}</AssemblySearchPaths>
```

```suggestion
    <AssemblySearchPaths Condition="'$(AssemblySearchPath_UseRawFileName)' != 'false'">$(AssemblySearchPaths);{RawFileName}</AssemblySearchPaths>
```

**CHANGES**

- Use quotes to avoid erroring out with a malformed value (_with space and illegal symbols_) entering the condition.
- Wouldn't it better to use `BaseProperty_SubOption` with an `_` acting as a separator to make these properties clearer?

See that we already use this pattern of naming here...

https://github.com/dotnet/msbuild/blob/14313d17e407e24ad80e113bd475d1dd7c698656/src/Tasks/Microsoft.Common.props#L181

---

I added the quotes. I think your BaseProperty_SubOption suggestion probably would've been a bit clearer, but I'm not super interested in changing the code we merged ~4 weeks ago, so I'm voting to leave it as-is on that point.

---

This change is not yet shipped with MSBuild, right? Meaning, it's not too late to change it. I could open a PR there if it's okay with you?

---

I'm fine with that. You're right that we haven't shipped it yet, but a lot of people are off right now for the holidays. I won't promise it'll get in, but if you make it, I'll approve, and if one more person approves early enough, we can merge it. Sound reasonable?

Make sure to update the docs, too.

---

I have made the changes. Here's the PR dotnet/msbuild#7166.